lstmcpipe/utils.py

```diff
@@ -0,0 +1,30 @@
+from pathlib import Path
+import subprocess
+
+
+def rerun_cmd(cmd, outfile, max_ntry=2):
+    """
+    rerun r0_to_dl1 process given by `cmd` as long as the exit code is 0 and number of try < max_ntry
+    move the failed output file to subdir failed_outputs
+
+    Parameters
+    ----------
+    cmd: str
+    simtel_file: Path
+    outfile: Path
+        path to the cmd output file
+    max_ntry: int
+    """
+    outfile = Path(outfile)
+    ret = -1
+    ntry = 1
+    while ret != 0 and ntry <= max_ntry:
+        ret = subprocess.run(cmd)
+        if ret.returncode != 0:
+            failed_jobs_subdir = Path(outfile.parent).joinpath('failed_outputs')
+            if outfile.exists():
+                failed_jobs_subdir.mkdir(exist_ok=True)
+                outfile_target = failed_jobs_subdir.joinpath(outfile.name)
+                print(f"Move failed output file from {outfile} to {outfile_target}. try #{ntry}")
+                outfile.rename(outfile_target)
```

Comment:
Why not using a `shutil.move` ?

Reply:
I think `rename` does not work across file system boundaries, so using `shutil.move` might be needed for some specific cases:

```
In [2]: p = Path('foo')

In [3]: p.rename('/tmp/bar')
---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
Input In [3], in <cell line: 1>()
----> 1 p.rename('/tmp/bar')

File /usr/lib/python3.10/pathlib.py:1232, in Path.rename(self, target)
   1222 def rename(self, target):
   1223     """
   1224     Rename this path to the target path.
   1225 
   (...)
   1230     Returns the new Path instance pointing to the target path.
   1231     """
-> 1232     self._accessor.rename(self, target)
   1233     return self.__class__(target)

OSError: [Errno 18] Invalid cross-device link: 'foo' -> '/tmp/bar'
```